validator/src/main.rs

```diff
@@ -641,23 +697,40 @@ pub fn main() {
             &udp_sockets,
         );
 
-        let genesis_hash =
-            initialize_ledger_path(cluster_entrypoint, &ledger_path, no_snapshot_fetch)
+        if let Ok((rpc_addr, rpc_client)) = create_rpc_client(cluster_entrypoint) {
+            if !validator_config.voting_disabled {
+                check_vote_account(
+                    &rpc_client,
+                    &vote_account,
+                    &voting_keypair.pubkey(),
+                    &identity_keypair.pubkey(),
+                )
                 .unwrap_or_else(|err| {
-                    error!("Failed to download ledger: {}", err);
+                    error!("Failed to check vote account: {}", err);
                     exit(1);
                 });
+            }
 
-        if let Some(expected_genesis_hash) = validator_config.expected_genesis_hash {
-            if expected_genesis_hash != genesis_hash {
-                error!(
-                    "Genesis hash mismatch: expected {} but local genesis hash is {}",
-                    expected_genesis_hash, genesis_hash,
-                );
-                exit(1);
+            let genesis_hash =
+                initialize_ledger_path(&rpc_addr, &rpc_client, &ledger_path, no_snapshot_fetch)
+                    .unwrap_or_else(|err| {
+                        error!("Failed to download ledger: {}", err);
+                        exit(1);
+                    });
+
+            if let Some(expected_genesis_hash) = validator_config.expected_genesis_hash {
+                if expected_genesis_hash != genesis_hash {
+                    error!(
+                        "Genesis hash mismatch: expected {} but local genesis hash is {}",
+                        expected_genesis_hash, genesis_hash,
+                    );
+                    exit(1);
+                }
             }
+            validator_config.expected_genesis_hash = Some(genesis_hash);
+        } else {
+            error!("unable to create rpc client");
```

Comment:
Can your print the error itself along side the `unable to create rpc client` message?  It looks like we just consume the error and don't show the user?

Reply:
Nice catch! https://github.com/solana-labs/solana/pull/6790/files#diff-1f422d7e73618aca7bce9935356de012R725